### PR TITLE
Issue 2453: Enable getStreamInfo api for BatchClient.

### LINKED ISSUE #2453: Enable getStreamInfo api in BatchClient.
**Problem description**

Enable `getStreamInfo` api in the batch client with retrieves information about the stream.
The attributes include.
-  Stream name.
-  Stream scope.
-  StreamCut pointing to the current HEAD of the stream.
-  StreamCut pointing to the current TAIL of the stream.
-  Stream Seal Status.
-  Stream Creation time.
- Stream end Time (! to be reviewed.)

**Problem location**
BatchClient.

**Suggestions for an improvement**
Implement the API.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**
Enable `BatchClient#getStreamInfo` API in batchclient. This api returns the current TAIL and current HEAD of the stream.

**Purpose of the change**
Fix issue #2453 

**What the code does**
getStreamInfo now returns 
* Stream name.
* Stream scope.
* StreamCut pointing to the current HEAD of the stream.
* StreamCut pointing to the current TAIL of the stream.

**How to verify it**
All existing tests should pass, new tests have been added to cover the new API.

### CODE REVIEW

**Comment by shrids** [comment]
[Enable getStreamInfo api for BatchClient](https://trello.com/c/PiT6YmPL/39-enable-getstreaminfo-api-for-batchclient)

File: client/src/main/java/io/pravega/client/batch/StreamInfo.java [hunk 1]
```diff
@@ -10,17 +10,37 @@
...
+ * This class is used to represent Stream information. It currently includes:
+ *
+ *  <li> - scope of stream.
+ *  <li> - name of stream.
+ *  <li> - {@link StreamCut} which represents the current TAIL of the stream.
+ *  <li> - {@link StreamCut} which represents the current HEAD of the stream.
+ */
 @Beta
 @Data
 public class StreamInfo {
```
Comment (tkaitchuck) [inline client/src/main/java/io/pravega/client/batch/StreamInfo.java]: We still have creationTime and lastUpdated time that we could be returning. Is that going to be in a separate PR?

Comment (shrids) [reply client/src/main/java/io/pravega/client/batch/StreamInfo.java]: yes, will raise it as a separate PR.
